source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -648,42 +650,46 @@ see the following API Documentation:
 - `TypeWithTypeParameters <{+api+}/apidocs/bson/org/bson/codecs/pojo/TypeWithTypeParameters.html>`__
 
 For more information on generics and type parameters, see the
-:`Oracle documentation on Invoking and Instantiating a Generic Type <https://docs.oracle.com/javase/tutorial/java/generics/types.html#instantiation>`__.
+`Java language guide on Invoking and Instantiating a Generic Type <https://docs.oracle.com/javase/tutorial/java/generics/types.html#instantiation>`__.
 
 Enum Type Support
 ^^^^^^^^^^^^^^^^^
 
-The POJO ``Codec`` uses the ``enum`` constant name as the property value by
-default. The following code snippet shows a POJO that contains an ``enum``
-field:
-
-.. code-block:: java
+The ``PojoCodecProvder`` includes a codec to convert the ``enum`` property
+type field in your POJO. The codec uses the name of the ``enum`` as the
+property value by default.
 
-   public enum Family {
-       ADELIE,
-       EMPEROR,
-       GENTOO,
-       ROCKHOPPER
-   }
+.. note::
 
-   public class Penguin {
+   When using driver versions 4.5 and later, you can include the ``EnumCodec``
+   without the ``PojoCodecProvider``.
 
-       private String name;
-       private Integer height;
-       private Family family;
+The following code snippets show an example ``enum`` and POJO that contains
+that ``enum`` field:
 
-       // ...
-  }
+.. literalinclude:: /includes/fundamentals/code-snippets/codecs/Family.java
+   :language: java
+   :dedent:
+   :start-after: start familyEnum
+   :end-before: end familyEnum
 
-The following snippets show an example instantiation of the ``Penguin``
-class and the resulting document when using the default behavior of the
-``EnumCodec`` included in the ``PojoCodecProvider``:
+.. literalinclude:: /includes/fundamentals/code-snippets/codecs/Penguin.java
+   :language: java
+   :dedent:
+   :emphasize-lines: 4
+   :start-after: start penguinClass
+   :end-before: end penguinClass
 
-.. code-block:: java
+The following code example shows how you can set the registry and insert
+an instance of the ``Penguin`` POJO with an ``enum`` field:
 
-   Penguin penguin = new Penguin("Robin", 51, ADELIE);
+.. literalinclude:: /includes/fundamentals/code-snippets/codecs/EnumFieldPojo.java
+   :language: java
+   :dedent:
+   :start-after: start createPenguin
+   :end-before: end createPenguin
 
-The resulting document should look something like this:
+The resulting document should contain the following data:
```

Comment:
Should this be an io codeblock?

Reply:
No, it doesn't show output. It's describing the document structure, so JSON would be the best option.